core/filter_api.php

```diff
@@ -1105,6 +1118,45 @@ function filter_get_query_sort_data( array &$p_filter, $p_show_sticky, array $p_
 				}
 			}
 
+		# if sorting by foreign key column
+		} else if ( array_key_exists( $c_sort, $s_fk_map ) ) {
+			$t_fk_table_alias = $s_fk_map[$c_sort]['table'] . '_' . $c_sort;
+			
+			# Defining sort field for user fields
+			# The if statement below is a hack required to have (at least semi-)
+			# proper sorting of username fields (see #10853); the code somewhat
+			# replicates the logic from function user_get_name()
+			# use of CASE, COALESCE should be fine for all db (SQL92 standard)
+			# however string concatenation operator '||' has a different meaning
+			# in MySQL, so we use CONCAT function which exists in all DBs
+			if( !array_key_exists( 'sort_field', $s_fk_map[$c_sort] ) ) {
```

Comment:
it's assuming that a non existant sort_field in the array is always a user table. Maybe we need to extend this in the future, so checking for the actual sort criteria is good
.